OpenSim/Simulation/Model/AbstractTool.cpp

```diff
@@ -597,48 +604,48 @@ bool AbstractTool::createExternalLoads( const string& aExternalLoadsFileName, Mo
             temp = new Storage(loadKinematicsFileName);
             if(!temp){
                 IO::chDir(savedCwd);
-                throw Exception("DynamicsTool: could not find external loads kinematics file '"+loadKinematicsFileName+"'."); 
+                throw Exception("AbstractTool: could not find external loads kinematics file '"+loadKinematicsFileName+"'."); 
             }
         }
         // if loading the data, do whatever filtering operations are also specified
-        if(temp && _externalLoads.getLowpassCutoffFrequencyForLoadKinematics() >= 0) {
-            cout<<"\n\nLow-pass filtering coordinates data with a cutoff frequency of "<<_externalLoads.getLowpassCutoffFrequencyForLoadKinematics()<<"."<<endl;
+        if(temp && externalLoads->getLowpassCutoffFrequencyForLoadKinematics() >= 0) {
+            cout<<"\n\nLow-pass filtering coordinates data with a cutoff frequency of "
+                << externalLoads->getLowpassCutoffFrequencyForLoadKinematics() <<"."<<endl;
             temp->pad(temp->getSize()/2);
-            temp->lowpassIIR(_externalLoads.getLowpassCutoffFrequencyForLoadKinematics());
+            temp->lowpassIIR(externalLoads->getLowpassCutoffFrequencyForLoadKinematics());
         }
         loadKinematicsForPointTransformation = temp;
     }
     
     // if load kinematics for performing re-expressing the point of application is provided
     // then perform the transformations
     if(loadKinematicsForPointTransformation){
-        SimTK::State& s = aModel.initSystem();
-        
+        SimTK::State& s = copyModel.initSystem();
+
         // Form complete storage so that the kinematics match the state labels/ordering
         Storage *qStore=NULL;
         Storage *uStore=NULL;
-        aModel.getSimbodyEngine().formCompleteStorages(s, 
+        copyModel.getSimbodyEngine().formCompleteStorages(s,
             *loadKinematicsForPointTransformation,
-            qStore,
-            uStore);
-       
-        _externalLoads.transformPointsExpressedInGroundToAppliedBodies(*qStore, _ti, _tf);
+            qStore, uStore);
+
+        externalLoads->transformPointsExpressedInGroundToAppliedBodies(*qStore, _ti, _tf);
         delete qStore;
         delete uStore;
-
-        aModel.invalidateSystem();
-    }
-    
-    // Add external loads to the set of all model forces
-    for(int i=0; i<_externalLoads.getSize(); ++i){
-        aModel.addForce(&_externalLoads[i]);
     }
 
+    //Now add the ExternalLoads (transformed or not) to the Model to be analyzed
+    ExternalLoads* exLoadsClone = externalLoads->clone();
+    aModel.addModelComponent(exLoadsClone);
+
+    // copy over created external loads to the external loads owned by the tool
+    _externalLoads = *externalLoads;
```

Comment:
I'm confused why this is necessary. Why is there an `ExternalLoads` in both the tool and the model?

Reply:
If this is how the code was before, then it's fine I suppose.